gnocchi/cli.py

```diff
@@ -162,7 +162,11 @@ def set_block(self, event):
         get_members_req = self._coord.get_members(self.GROUP_ID)
         try:
             members = sorted(get_members_req.get())
-            self.block_index = members.index(self._my_id)
+            try:
+                self.block_index = members.index(self._my_id)
+            except ValueError:
+                LOG.error('Unable to find myself in group, restarting')
```

Comment:
Maybe you should call self.terminate() to cleanly shutdown, or at least self._coord.stop() to disconnect from redis before raising SystemExit().